packages/story-editor/src/components/carousel/pagepreview/index.js

```diff
@@ -100,9 +100,19 @@ function PagePreview({ page, label, ...props }) {
   const [imageBlob, setImageBlob] = useState();
   const [pageNode, setPageNode] = useState();
   const setPageRef = useCallback((node) => node && setPageNode(node), []);
-  const hasImage = !isActive && imageBlob;
   const pageAtGenerationTime = useRef();
 
+  const activeStateTracker = useRef(isActive);
+  const isPageImageUnchanged = isActive && activeStateTracker.current === false;
+  const hasImage = (!isActive || isPageImageUnchanged) && imageBlob;
```

Comment:
The `isActive &&` here is unnecessary, as `isPageImageUnchanged` will only ever be looked at if `isActive` is not false (so _truthy_).

Also, is the `=== false` necessary? Could it be other _falsey_ values?

These could probably just be collapsed to:

```suggestion
  const hasImage = (!isActive || !activeStateTracker.current) && imageBlob;
```